config/routes.rb

```diff
@@ -2626,14 +2626,52 @@
   # OPTIONS requests for REST API pre-flight checks
 
   # Semantic Versioning Regex for API, i.e. vMajor.minor.patch[-pre]
-  apiver_regex = /v[\d]+(\.[\da-zA-Z]+)*(\-[\da-zA-Z]+)?/
+  API_VERSION_REGEX = /v[\d]+(\.[\da-zA-Z]+)*(\-[\da-zA-Z]+)?/
 
   match '/api/*path'   => 'api#handle_options_request', :via => [:options]
-  get '/api(/:version)'           => 'api#show',    :format => 'json', :version => apiver_regex
+  get '/api(/:version)' => 'api#show_entrypoint', :format => 'json', :version => API_VERSION_REGEX
+
+  API_ACTION_PREFIXES = {
+    :get    => "show",
+    :post   => "update",
+    :put    => "update",
+    :patch  => "update",
+    :delete => "destroy"
+  }.freeze
+
+  def action_for(verb, collection)
+    prefix = API_ACTION_PREFIXES[verb]
+
+    if ApiController.method_defined?("#{prefix}_#{collection}")
```

Comment:
@imtayadeway and I found the slowness was actually loading of ApiController that was the problem.  Tim is looking for a way to avoid this, which involves temporarily reverting some of his deletions with the hope of a followup PR to clean it up more completely.  👍